main.py

```diff
@@ -0,0 +1,63 @@
+import os
+from skimage.io import imread
+from skimage.transform import resize
+import numpy
+from sklearn.decomposition import PCA
+from sklearn.model_selection import train_test_split
+from sklearn.svm import SVC
+from sklearn.model_selection import GridSearchCV
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.metrics import accuracy_score
+
+training_path = os.path.join(__file__, '..\sample-images\\training-images')
+
+data = []
+labels = []
+# Fetch the image data from the training folder
+for folder in os.listdir(training_path):
+    for file in os.listdir(os.path.join(training_path, folder)):
+        image_path = os.path.join(training_path, folder, file)
+        image = imread(image_path)
+        resized_image = resize(image, (100, 100))
+        image_data = resized_image.flatten()
+        data.append(image_data)
+        labels.append(folder)
+
+image_data = numpy.asarray(data)
+image_labels = numpy.asarray(labels)
+
+# Reduce dimensions of training and test data with Principal Component Analysis
+pca = PCA(n_components=10)
```

Comment:
Get rid of the magic number, also is 10 best? We could spend some time trying to find the optimal number of components but it doesn't really matter for this project.